cluster-autoscaler/cloudprovider/huaweicloud/huaweicloud_cloud_provider.go

```diff
@@ -176,6 +161,38 @@ func (hcp *huaweicloudCloudProvider) Refresh() error {
 	return nil
 }
 
+func (hcp *huaweicloudCloudProvider) buildAsgs(specs []string) error {
+	for _, spec := range specs {
+		if err := hcp.addNodeGroup(spec); err != nil {
+			klog.Warningf("failed to add node group to huaweicloud provider with spec: %s", spec)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (hcp *huaweicloudCloudProvider) addNodeGroup(spec string) error {
+	asgs, err := hcp.cloudServiceManager.ListScalingGroups()
```

Comment:
One request for each spec seems a little bit expensive if the user specifies many specs. 
How do you think?

Reply:
Make sense, I will move it to upper function.